### PR TITLE
Jv/3.6.x fix build

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

See above

### CODE REVIEW

**Comment by JoukoVirtanen** [comment]
This branch was effectively copied from https://github.com/stackrox/collector/pull/698

which cherrypicked the following

Bump builder base image to centos stream
Remove 'Development Tools' from builder image

and also removed the prepare legacy sources step.

In addition fcb8366 Re-enable "CPaaS" ubi-image job was cherrypicked and conflicts were resolved.

File: .circleci/dockerized-drivers/build-drivers.sh [hunk 1]
```diff
@@ -31,14 +31,14 @@ if [[ -z "$(ls -A ~/workspace/go/src/github.com/stackrox/bundles)" ]]; then
...
-	-f "${SOURCE_ROOT}/kernel-modules/dockerized/Dockerfile" \
-	~/workspace/go/src/github.com/stackrox
+    --build-arg BRANCH="${GIT_REF}" \
+    --build-arg REDHAT_SUBSCRIPTION_ORG_ID="${REDHAT_SUBSCRIPTION_ORG_ID}" \
+    --build-arg REDHAT_SUBSCRIPTION_ACTIVATION_KEY="${REDHAT_SUBSCRIPTION_ACTIVATION_KEY}" \
+    --build-arg CACHE_REPO="${DRIVER_REPO}" \
+    --build-arg CACHE_TAG="${COLLECTOR_DRIVERS_CACHE}" \
+    --tag "${IMAGE_TAG}" \
+    -f "${SOURCE_ROOT}/kernel-modules/dockerized/Dockerfile" \
+    "${STACKROX_ROOT}"
```
Comment (robbycochran) [inline .circleci/dockerized-drivers/build-drivers.sh:41]: I belive the `STACKROX_ROOT` var is undefined. However, the dockerized steps are disabled in the branch so it won't be executed.

File: .circleci/prepare-kernels/prepare-legacy-sources.sh [hunk 2]
```diff
@@ -4,6 +4,11 @@ set -eo pipefail
 TAG=$1
 BRANCH=$2
 
+if [[ -z "$TAG" && "$BRANCH" != "master" && ! -f "${WORKSPACE_ROOT}/pr-metadata/labels/build-legacy-probes" ]]; then
```
Comment (robbycochran) [inline .circleci/prepare-kernels/prepare-legacy-sources.sh:7]: We should add this change on the master branch as well if it is applicable.

**Review by robbycochran — APPROVED** [review]
LGTM. Please describe all changes in the commit message.

File: .circleci/prepare-kernels/prepare-legacy-sources.sh (hunk 2 above)
Comment (JoukoVirtanen) [reply .circleci/prepare-kernels/prepare-legacy-sources.sh:7]: I will look into it.

**Pull request merged**